src/features/defi/providers/foxy/components/FoxyManager/Deposit/FoxyDeposit.tsx

```diff
@@ -0,0 +1,661 @@
+  const balance = useAppSelector(state => selectPortfolioCryptoBalanceByAssetId(state, assetCAIP19))
+  const loading = useSelector(selectPortfolioLoading)
+
+  // notify
+  const toast = useToast()
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        if (!walletState.wallet || !contractAddress) return
+        const [address, foxyOpportunity, pricePerShare] = await Promise.all([
+          chainAdapter.getAddress({ wallet: walletState.wallet }),
+          api.getFoxyOpportunityByStakingAddress(contractAddress),
+          api.pricePerShare()
+        ])
+        dispatch({ type: FoxyDepositActionType.SET_USER_ADDRESS, payload: address })
+        dispatch({ type: FoxyDepositActionType.SET_OPPORTUNITY, payload: foxyOpportunity })
+        dispatch({
+          type: FoxyDepositActionType.SET_PRICE_PER_SHARE,
+          payload: pricePerShare.toString()
+        })
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('FoxyDeposit error:', error)
+      }
+    })()
+  }, [api, chainAdapter, contractAddress, walletState.wallet])
+
+  const getApproveGasEstimate = async () => {
+    if (!state.userAddress || !tokenId) return
+    try {
+      const [gasLimit, gasPrice] = await Promise.all([
+        api.estimateApproveGas({
+          tokenContractAddress: tokenId,
+          contractAddress,
+          userAddress: state.userAddress
+        }),
+        api.getGasPrice()
+      ])
+      return bnOrZero(gasPrice).times(gasLimit).toFixed(0)
+    } catch (error) {
+      console.error('FoxyDeposit:getApproveEstimate error:', error)
+      toast({
+        position: 'top-right',
+        description: translate('common.somethingWentWrongBody'),
+        title: translate('common.somethingWentWrong'),
+        status: 'error'
+      })
+    }
+  }
+
+  const getDepositGasEstimate = async (deposit: DepositValues) => {
+    if (!state.userAddress || !tokenId) return
+    try {
+      const [gasLimit, gasPrice] = await Promise.all([
+        api.estimateDepositGas({
+          // tokenContractAddress: tokenId,
+          tokenContractAddress: tokenId,
+          contractAddress,
+          amountDesired: bnOrZero(deposit.cryptoAmount)
+            .times(`1e+${asset.precision}`)
+            .decimalPlaces(0),
+          userAddress: state.userAddress
+        }),
+        api.getGasPrice()
+      ])
+      return bnOrZero(gasPrice).times(gasLimit).toFixed(0)
+    } catch (error) {
+      console.error('FoxyDeposit:getDepositGasEstimate error:', error)
+      toast({
+        position: 'top-right',
+        description: translate('common.somethingWentWrongBody'),
+        title: translate('common.somethingWentWrong'),
+        status: 'error'
+      })
+    }
+  }
+
+  const handleContinue = async (formValues: DepositValues) => {
+    if (!state.userAddress) return
+    // set deposit state for future use
+    dispatch({ type: FoxyDepositActionType.SET_DEPOSIT, payload: formValues })
```

Comment:
i'm not sure this is necessary, try removing it, it looks like we're setting this prematurely before doing validation?